src/plugins/SectionCapsPlugin/SectionCapsPlugin.js

```diff
@@ -0,0 +1,362 @@
+import {Plugin} from "../../viewer/Plugin.js";
+import {math} from "../../viewer/scene/math/math.js";
+import { Mesh } from "../../viewer/scene/mesh/Mesh.js";
+import { ReadableGeometry } from "../../viewer/scene/geometry/ReadableGeometry.js";
+import { PhongMaterial } from "../../viewer/scene/materials/PhongMaterial.js";
+import CSG from "../lib/csg.js";
+
+class SectionCapsPlugin extends Plugin {
+    constructor(viewer, cfg = {}) {
+        super("SectionCaps", viewer, cfg);
+        this.sectionPlanes = [];
+        // this.sceneModel = viewer.scene.models.myModel;
+        this.sceneModel = Object.keys(viewer.scene.models).map((key) => viewer.scene.models[key]);
+        this.prevIntersectionModels = {};
+        this.posTimeout = null;
+        this.dirTimeout = null;
+        this.setupSectionPlanes();
+    }
+
+
+    //this function will be used to setup event listeners for creation of section planes
+    setupSectionPlanes(){
```

Comment:
Could we please make any methods private that are only used by SectionCapsPlugin

Reply:
Actually, it looks as if almost all methods and variables on SectionCapsPlugin can be made private, since it effectively works in the background. 

I would add something like a `setEnabled(enabled)` method to SectionCapsPlugin, though, so that the user can instantiate the plugin, but still make it inactive temporarily.

Reply:
Done